micronaut-mcp-server-java-sdk/src/main/java/io/micronaut/mcp/server/exceptions/SerdeExceptionMcpErrorMapper.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.mcp.server.exceptions;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.serde.exceptions.SerdeException;
+import io.modelcontextprotocol.spec.McpError;
+import io.modelcontextprotocol.spec.McpSchema;
+import jakarta.inject.Singleton;
+
+/**
+ * {@link McpErrorExceptionMapper} for {@link SerdeException}.
+ */
+@Requires(classes = SerdeException.class)
+@Singleton
+@Internal
+class SerdeExceptionMcpErrorMapper implements McpErrorExceptionMapper<SerdeException> {
```

Comment:
can be `final`

Reply:
resolved via https://github.com/micronaut-projects/micronaut-mcp/pull/52/commits/0bd72a5cc71b65a309ab8a6d19587b9602145359